src/core/ghosts.cpp

```diff
@@ -29,29 +29,90 @@
 #include "errorhandling.hpp"
 #include "particle_data.hpp"
 
+#include <boost/serialization/vector.hpp>
+#include <mpi.h>
+#include <utils/Span.hpp>
+
 #include <algorithm>
 #include <cstdio>
-#include <cstdlib>
 #include <cstring>
-#include <mpi.h>
+#include <type_traits>
 #include <vector>
 
 /** Tag for communication in ghost_comm. */
 #define REQ_GHOST_SEND 100
 
-static int n_s_buffer = 0;
-static int max_s_buffer = 0;
-/** send buffer. Just grows, which should be ok */
-static char *s_buffer = nullptr;
+struct CommBuf {
```

Comment:
please add a docstring for the struct

Reply:
also think about using ```class``` instead of ```struct``` here. [core guideline](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c8-use-class-rather-than-struct-if-any-member-is-non-public)